### PR TITLE
[zh-cn] update Using_templates_and_slots

### PR DESCRIPTION
**Pull request by YieldRay**
### Description

Also fix some typos and sync English content.

### Additional details

Source: <https://github.com/mdn/content/blob/main/files/en-us/web/api/web_components/using_templates_and_slots/index.md>

### CODE REVIEW

File: files/zh-cn/web/api/web_components/using_templates_and_slots/index.md [hunk 1]
```diff
@@ -186,20 +184,20 @@ customElements.define(
...
-- {{HTMLElement("template")}} 用 `<slot>` 和它的 [`name`](/zh-CN/docs/Web/HTML/Element/slot#name) 属性生成一个 [命名的槽 (named slots)](/zh-CN/docs/Web/HTML/Element/slot#named-slot)：
+- {{HTMLElement("template")}} 标签有一个 {{HTMLElement("style")}} 标签，里面有一些只能在当前 {{HTMLElement("template")}} 中创建的文档片段范围内生效的 CSS 样式
+- {{HTMLElement("template")}} 用 {{HTMLElement("slot")}} 和它的 [`name`](/zh-CN/docs/Web/HTML/Element/slot#name) 属性生成一个[命名插槽](/zh-CN/docs/Web/HTML/Element/slot#named-slot)：
 
   - `<slot name="element-name">`
   - `<slot name="description">`
   - `<slot name="attributes">`
 
-- {{HTMLElement("template")}} 包裹 [命名的槽 (named slots)](/zh-CN/docs/Web/HTML/Element/slot#named-slot) 在 {{HTMLElement("details")}} 元素中。
+- {{HTMLElement("details")}} 元素中的 {{HTMLElement("template")}} 包裹[命名插槽](/zh-CN/docs/Web/HTML/Element/slot#named-slot)。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/web_components/using_templates_and_slots/index.md:196]:
```suggestion
- {{HTMLElement("template")}} 将[命名插槽](/zh-CN/docs/Web/HTML/Element/slot#命名插槽)包裹在 {{HTMLElement("details")}} 元素中。
```

File: files/zh-cn/web/api/web_components/using_templates_and_slots/index.md [hunk 2]
```diff
@@ -186,20 +184,20 @@ customElements.define(
...
 
-### 从\<template> 中创建一个新的 \<element-details> 对象
+### 从 \<template> 中创建一个新的 \<element-details> 元素
 
-接下来，我们定义一个新的 **`<element-details>`** 元素然后用 {{DOMXref("Element.attachShadow")}} 来将它附着到它的 [阴影 (shadow) root](/zh-CN/docs/Web/API/ShadowRoot), 这个阴影上我们创建了 {{HTMLElement("template")}} 元素。这和我们上面使用的简单例子中的一样。
+接下来，我们定义一个新的 **`<element-details>`** 元素然后用 {{DOMXref("Element.attachShadow")}} 来将其附加到它的 [影子根](/zh-CN/docs/Web/API/ShadowRoot) 上，即通过上面的 {{HTMLElement("template")}} 元素创建的文档片段。这和我们上面使用的简单例子中的一样。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/web_components/using_templates_and_slots/index.md:200]:
```suggestion
接下来，我们创建一个名为 **`<element-details>`** 的新自定义元素，然后用 {{DOMXref("Element.attachShadow")}} 来将上面通过 {{HTMLElement("template")}} 元素创建的文档片段作为[影子根](/zh-CN/docs/Web/API/ShadowRoot)附加到它上面，这使用了与我们在之前的简单示例中看到的完全相同的模式。
```

File: files/zh-cn/web/api/web_components/using_templates_and_slots/index.md [hunk 3]
```diff
@@ -245,15 +243,15 @@ customElements.define(
...
 
-关于以上代码片段，请注意以下几点：
+关于以上代码片段，注意以下几点：
 
-- 该代码片段有 2 个 **`<element-details>`** 标签，他们都使用了“slot”属性来引用名为`"element-name"`和`"description"`的槽，并把他们都放在根下。
-- 只有第一个 **`<element-details>`** 元素 引用了名为 `"attributes"` 的槽，而第二个 `<element-details>` 标签没有引用名为`"attributes"`的槽。
-- 只有第一个 **`<element-details>`** 标签中的"dl"使用了名为 `"attributes"`的槽，他包含子元素："dt"和"dd"元素。
+- 该代码片段有两个 **`<element-details>`** 标签，它们都使用了 [`slot`](/zh-CN/docs/Web/HTML/Global_attributes#slot) 属性来引用名为 `"element-name"` 和 `"description"` 的插槽，并把它们都放在影子根下。
+- 只有第一个 **`<element-details>`** 元素引用了名为 `"attributes"` 的插槽，而第二个 `<element-details>` 标签没有引用名为 `"attributes"` 的插槽。
+- 只有第一个 **`<element-details>`** 标签中的 {{HTMLElement("dl")}} 元素使用了名为 `"attributes"` 的插槽，它包含子元素：{{HTMLElement("dt")}} 和 {{HTMLElement("dd")}} 元素。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/web_components/using_templates_and_slots/index.md:250]:
```suggestion
- 该代码片段有两个 **`<element-details>`** 元素，它们都使用了 [`slot`](/zh-CN/docs/Web/HTML/Global_attributes#slot) 属性来引用名为 `"element-name"` 和 `"description"` 的[命名插槽](/zh-CN/docs/Web/HTML/Element/slot#命名插槽)，并把它们都放在[影子根](/zh-CN/docs/Web/API/ShadowRoot)下。
- 只有第一个 **`<element-details>`** 元素引用了名为 `"attributes"` 的[命名插槽](/zh-CN/docs/Web/HTML/Element/slot#命名插槽)，而第二个 `<element-details>` 标签没有引用名为 `"attributes"` 的[命名插槽](/zh-CN/docs/Web/HTML/Element/slot#命名插槽)。
- 第一个 `<element-details>` 元素中的 {{HTMLElement("dl")}} 元素使用包含子元素 {{HTMLElement("dt")}} 和 {{HTMLElement("dd")}} 的 {{HTMLElement("dl")}} 元素引用了 `"attributes"` [命名插槽](/zh-CN/docs/Web/HTML/Element/slot#命名插槽)。
```

File: files/zh-cn/web/api/web_components/using_templates_and_slots/index.md [hunk 4]
```diff
@@ -245,15 +243,15 @@ customElements.define(
...
 
 ### 添加一些最终的样式
 
-在完成之前，我们在文档中给"dl"和"dt"以及"dd"标签增加一些 CSS 样式。
+在完成之前，我们在文档中给 {{HTMLElement("dl")}} 和 {{HTMLElement("dt")}} 以及 {{HTMLElement("dd")}} 标签增加一些 CSS 样式。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/web_components/using_templates_and_slots/index.md:254]:
```suggestion
在完成之前，我们在文档中给 {{HTMLElement("dl")}} 和 {{HTMLElement("dt")}} 以及 {{HTMLElement("dd")}} 元素增加一些 CSS 样式。
```

**Review by JasonLamv-t — CHANGES_REQUESTED** [review]
Sorry for the late review. There are some suggestions which already mention, and I want to point out additional:
- please translate the string or text content wrapped in the element of code examples - if it will not fuzzy or distort the purpose of the example;
- I tend to not to mix the terms `标签` and `元素`, please using `元素` for consistency

File: files/zh-cn/web/api/web_components/using_templates_and_slots/index.md [hunk 5]
```diff
@@ -186,20 +184,20 @@ customElements.define(
 </template>
 ```
 
-上面这个{{HTMLElement("template")}} 有几个特征：
+上面这个 {{HTMLElement("template")}} 元素有几个特征：
 
-- {{HTMLElement("template")}} 标签有一个 {{HTMLElement("style")}} 标签，里面有一些 只能在当前{{HTMLElement("template")}} 的派生中生效的 CSS 样式
-- {{HTMLElement("template")}} 用 `<slot>` 和它的 [`name`](/zh-CN/docs/Web/HTML/Element/slot#name) 属性生成一个 [命名的槽 (named slots)](/zh-CN/docs/Web/HTML/Element/slot#named-slot)：
+- {{HTMLElement("template")}} 标签有一个 {{HTMLElement("style")}} 标签，里面有一些只能在当前 {{HTMLElement("template")}} 中创建的文档片段范围内生效的 CSS 样式
+- {{HTMLElement("template")}} 用 {{HTMLElement("slot")}} 和它的 [`name`](/zh-CN/docs/Web/HTML/Element/slot#name) 属性生成一个[命名插槽](/zh-CN/docs/Web/HTML/Element/slot#named-slot)：
```
[the remark this answers is not in the document]
Comment (yin1999) [reply files/zh-cn/web/api/web_components/using_templates_and_slots/index.md:190]: 建议用“具名插槽”这一翻译

File: files/zh-cn/web/api/web_components/using_templates_and_slots/index.md (hunk 1 above)
Comment (yin1999) [reply files/zh-cn/web/api/web_components/using_templates_and_slots/index.md:196]: +1

File: files/zh-cn/web/api/web_components/using_templates_and_slots/index.md (hunk 3 above)
Comment (yin1999) [reply files/zh-cn/web/api/web_components/using_templates_and_slots/index.md:250]: +1

File: files/zh-cn/web/api/web_components/using_templates_and_slots/index.md [hunk 6]
```diff
@@ -68,29 +68,27 @@ customElements.define(
...
 
 ```html
 <my-paragraph></my-paragraph>
 ```
 
-> **备注：** 模板在浏览器中的支持情况很好，Firefox（版本 63 及以上），Chrome，Opera ,Edge（版本 79 及以上）和 Safari 支持 Shadow DOM
+## 使用插槽（slots）添加灵活度
 
-## 使用槽 (slots) 添加灵活度
+尽管到这一步已经挺好了，但是元素仍旧不是很灵活。我们只能在里面放一点文本，甚至没有普通的 p 标签管用！我们可以使用 {{htmlelement("slot")}} 元素让它能在单个实例中通过声明式的语法展示不同的文本。
```
[the remark this answers is not in the document]
Comment (YieldRay) [reply files/zh-cn/web/api/web_components/using_templates_and_slots/index.md]: I find the existing translation did not distinguish between `markup` and `element`, will fix it.

File: files/zh-cn/web/api/web_components/using_templates_and_slots/index.md [hunk 7]
```diff
@@ -68,29 +68,27 @@ customElements.define(
...
 
-尽管到这一步已经挺好了，但是元素仍旧不是很灵活。我们只能在里面放一点文本，甚至没有普通的 p 标签管用！我们使用 {{htmlelement("slot")}} 让它能在单个实例中通过声明式的语法展示不同的文本。浏览器对这个功能的支持比{{htmlelement("template")}}少，在 Chrome 53, Opera 40, Safari 10，火狐 59 和 Edge 79 中支持。
+插槽由其 `name` 属性标识，并且允许你在模板中定义占位符，当在标记中使用该元素时，该占位符可以填充所需的任何 HTML 标记片段。
 
-插槽由其`name`属性标识，并且允许你在模板中定义占位符，当在标记中使用该元素时，该占位符可以填充所需的任何 HTML 标记片段。
-
-如果你想添加一个槽到我们的这个例子，我们会将模板的 p 标签改成下面这样：
+因此，如果我们想添加一个插槽到这个简单的例子中，我们会将模板的 p 标签改成下面这样：
```
[the remark this answers is not in the document]
Comment (YieldRay) [reply files/zh-cn/web/api/web_components/using_templates_and_slots/index.md]: Is `段落元素` fine here? As the original text is `paragraph element`.

File: files/zh-cn/web/api/web_components/using_templates_and_slots/index.md [hunk 8]
```diff
@@ -64,37 +66,35 @@ customElements.define(
...
 
-现在我们将它添加自己的 HTML 文档中试试看：
+现在我们将它添加到自己的 HTML 文档中试试看：
 
 ```html
 <my-paragraph></my-paragraph>
 ```
 
-> **备注：** 模板在浏览器中的支持情况很好，Firefox（版本 63 及以上），Chrome，Opera ,Edge（版本 79 及以上）和 Safari 支持 Shadow DOM
+## 使用插槽添加灵活度
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/web_components/using_templates_and_slots/index.md:79]:
```suggestion
## 使用插槽增加灵活度
```

**Review by JasonLamv-t — APPROVED** [review]
LGTM. Thank you, @YieldRay! 👍🏼

**Pull request merged**